packages/backend-function/src/factory.ts

```diff
@@ -234,3 +293,31 @@ const nodeVersionMap: Record<NodeVersion, Runtime> = {
   18: Runtime.NODEJS_18_X,
   20: Runtime.NODEJS_20_X,
 };
+
+const secretPathSuffix = '_PATH';
+const secretPlaceholderText = '<value will be resolved during runtime>';
+const secretPaths: string[] = [];
+
+const translateEnvironmentProp = (
+  functionEnvironmentProp: HydratedFunctionProps['environment'],
+  backendSecretResolver: BackendSecretResolver
+): Record<string, string> => {
+  const result: Record<string, string> = {};
+  const secretPathEnvVars = [];
+
+  for (const [key, value] of Object.entries(functionEnvironmentProp)) {
+    if (typeof value !== 'string') {
+      const secretPath = backendSecretResolver.resolveToPath(value);
+      result[key + secretPathSuffix] = secretPath;
+      result[key] = secretPlaceholderText;
+      secretPathEnvVars.push(key + secretPathSuffix);
+      secretPaths.push(secretPath);
+    } else {
+      result[key] = value;
+    }
+  }
+
+  process.env.SECRET_PATH_ENV_VARS = secretPathEnvVars.join(',');
```

Comment:
The comment above probably makes this a moot point, but functions should never communicate via environment variables. This should be part of the return value of the function if it is needed in other parts of the code.